pkg/operator/baremetal_config.go

```diff
@@ -132,11 +164,13 @@ func getIronicInspectorEndpoint(baremetalConfig BaremetalProvisioningConfig) *st
 }
 
 func getProvisioningDHCPRange(baremetalConfig BaremetalProvisioningConfig) *string {
```

Comment:
We talked about this on bluejeans, but I'll record a comment here for the next time I review this PR.

I think we want to do the field validation earlier in the process of reconciling the Provisioning CR, so that we don't reach a point where we want to use the ProvisioningDHCPRange if we know the value is invalid. That would mean we don't need the logic in this function, since we can always just use `&(baremetalConfig.ProvisioningDHCPRange)` and never have to worry about passing `nil` to whatever is calling this now.

Reply:
Yes, added a few checks earlier in the code when the provisioning config is read from the CR. These checks check for inter-dependencies between config values and reject the config at that point. This is different from the earlier method where we checked for the presence of config only to validate the CR and checked for inter-dependencies within the config before setting envVars. The new approach is better in that it detects bad config earlier.